src/stdlib/String.ts

```diff
@@ -264,8 +264,8 @@ export const Substitute = new Callable(
         },
         impl: (interpreter: Interpreter, str: BrsString, arg0: BrsString, arg1: BrsString, arg2: BrsString, arg3: BrsString): BrsString => {
             let tmpStr = str.value;
-            let replacements = [ arg0, arg1, arg2, arg3 ];
-            for (let index in replacements) {
```

Comment:
The `for ... in` lint check is enabled to avoid accidentally encountering the [weirdness of using it on arrays](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in#Array_iteration_and_for...in).  Manually creating the `replacements` object feels like a bit of a hack, because it seems like `for in` is what you wanted here. There's a few options:

1. Use `Array.prototype.forEach` to get both the index and value at the same time:
    ```typescript
    replacements.forEach( (replacement, index) => {
            let token = `{${index}}`;
            tmpStr = tmpStr.replace(new RegExp(token, "g"), replacement.value); 
    });
    ```
2. (My preference) Remove `tmpStr` entirely and use `Array.prototype.reduce`:
    ```typescript
        let completelyReplaced = [arg0, arg1, arg2, arg3].reduce(
            (replaced, replacement, index) => replaced.replace(new RegExp(`{${index}}`, "g"), replacement.value),
            str.value
        );
    ```
2. Disable TSLint on that line (please be stingy with this though :smile: )
    ```typescript
    // tslint:disable-next-line:no-for-in-array
    for (let index in replacements) {
        // ...
    }
    ```